src/test/java/com/artipie/maven/MavenITCase.java

```diff
@@ -23,187 +23,223 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
 import org.hamcrest.MatcherAssert;
-import org.hamcrest.core.AllOf;
-import org.hamcrest.core.StringContains;
-import org.hamcrest.text.MatchesPattern;
-import org.hamcrest.text.StringContainsInOrder;
+import org.hamcrest.Matchers;
+import org.hamcrest.TypeSafeMatcher;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.condition.DisabledOnOs;
 import org.junit.jupiter.api.condition.OS;
-import org.junit.jupiter.api.io.TempDir;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
-import org.testcontainers.shaded.org.apache.commons.io.FileUtils;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.utility.MountableFile;
 
 /**
  * Integration tests for Maven repository.
  * @since 0.11
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @todo #855:30min Deploy test is broken, mvn deploy fails
+ *  on 500 server respose due to file not exist exception at
+ *  maven upload.
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 @DisabledOnOs(OS.WINDOWS)
 final class MavenITCase {
 
     /**
-     * Temporary directory for all tests.
-     * @checkstyle VisibilityModifierCheck (3 lines)
+     * Artipie container.
      */
-    @TempDir
-    Path tmp;
+    private GenericContainer<?> artipie;
 
     /**
-     * Tested Artipie server.
+     * Client container.
      */
-    private ArtipieServer server;
+    private GenericContainer<?> client;
 
     /**
-     * Container.
+     * Network.
      */
-    private TestContainer cntn;
-
-    /**
-     * Storage.
-     */
-    private Storage storage;
-
-    /**
-     * Artipie server port.
-     */
-    private int port;
+    private Network net;
 
     @BeforeEach
     void init() throws Exception {
-        this.storage = new FileStorage(this.tmp);
-        this.server = new ArtipieServer(
-            this.tmp, "my-maven",
-            new RepoConfigYaml("maven").withFileStorage(this.tmp.resolve("repos"))
-        );
-        this.port = this.server.start();
-        new MavenSettings(this.port)
-            .writeTo(this.tmp);
-        this.cntn = new TestContainer("centos:centos8", this.tmp);
-        this.cntn.start(this.port);
-        this.cntn.execStdout("yum", "-y", "install", "maven");
+        this.net = Network.newNetwork();
+        this.artipie = new GenericContainer<>("artipie/artipie:1.0-SNAPSHOT")
+            .withNetwork(this.net).withNetworkAliases("artipie")
```

Comment:
@g4s8 I'd also suggest to todo for creating a class with artipie container, otherwise we will copy this code from test to test. Or we even could create a class with two containers - artipie and another one, this class can create network, containers, start and stop them.

Reply:
@olenagerasimova thanks, please check new commit